module-user/src/test/kotlin/com.example.domain.user/entity/StreamerUserTest.kt

```diff
@@ -0,0 +1,38 @@
+package com.example.domain.user.entity
+
+import org.assertj.core.api.Assertions.assertThat
+import org.junit.jupiter.api.Assertions
+import org.junit.jupiter.api.DisplayName
+import org.junit.jupiter.api.function.Executable
+import org.junit.jupiter.params.ParameterizedTest
+import org.junit.jupiter.params.provider.Arguments
+import org.junit.jupiter.params.provider.MethodSource
+import java.util.stream.Stream
+
+@DisplayName("[회원] 스트리머 회원")
+internal class StreamerUserTest {
+
+    companion object {
+        @JvmStatic
+        fun `user, streamingName으로 스트리머회원을 생성합니다`(): Stream<Arguments> {
+            return Stream.of(
+                Arguments.of(User("test@gmail.com", "mario"), "koopa")
+            )
+        }
+    }
+
+    @ParameterizedTest
+    @MethodSource
+    fun `user, streamingName으로 스트리머회원을 생성합니다`(user: User, streamingName: String) {
+        // given
+
+        // when
+        val streamer = StreamerUser(user, streamingName)
+
+        // then
+        Assertions.assertAll(
+            Executable { assertThat(streamer.user.status).isEqualTo(UserStatus.REGISTERED) },
+            Executable { assertThat(streamer.status).isEqualTo(StreamerUserStatus.PENDING) }
+        )
```

Comment:
```suggestion
        // then
        assertThat(streamer.user.status).isEqualTo(UserStatus.REGISTERED)
        assertThat(streamer.status).isEqualTo(StreamerUserStatus.PENDING)
```

코멘트를 수용하여 assertAll을 추가했습니다.
하지만 이게 묶기 전보다 가독성이 좋거나 보일러 플레이트가 적은가? 에 대해서는 여전히 의문이 있습니다.

좀 더 내용을 찾아보니 이렇게 보일러 플레이트가 생기는 이유는 `람다가 자동 캐스팅되지 않는 현재 코틀린의 한계`라고 합니다.

```
At present, there’s no support for automatically casting a lambda to a functional interface, so we need to do it by hand
```

[Multiple Assertions - baeldung](https://www.baeldung.com/kotlin/junit-5-kotlin#2-multiple-assertions)

Reply:
https://github.com/sirloin-dev/meatplatform-sandbox/blob/main/server/api-core/src/test/kotlin/testcase/small/domain/user/UserSpec.kt

이거 한번 보세요. 이정도면 꽤 읽을만 하지 않나요?

Reply:
어? `Executable` 안써도 되네요?
갑자기 깔끔해 보입니다. 이렇게 수정하겠습니다.